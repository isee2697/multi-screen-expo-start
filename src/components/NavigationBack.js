import React from 'react';
import { TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { utilStyles } from '../api/constants';

// grab svg icon
import SvgCircleLeft from './icons/Svg.CircleLeft';

const NavigationBack = props => {
  const { navigation } = props;

  return (
    <TouchableOpacity
      accessible
      accessibilityLabel="go back"
      accessibilityComponentType="button"
      accessibilityTraits="button"
      activeOpacity={utilStyles.opacityActive}
      onPress={() => navigation.goBack(navigation.state.key)}
      style={{ padding: 12 }}
    >
      <SvgCircleLeft active />
    </TouchableOpacity>
  );
};

// type checking
NavigationBack.propTypes = {
  // required
  navigation: PropTypes.object.isRequired
};

export default NavigationBack;
