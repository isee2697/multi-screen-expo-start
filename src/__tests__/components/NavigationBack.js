import React from 'react';
import { shallow } from 'enzyme';

import NavigationBack from '../../components/NavigationBack';

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  state: {
    key: 1
  }
};
const component = shallow(<NavigationBack navigation={navigation} />);

describe('<NavigationBack />', () => {
  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('onPress', () => {
    component.props().onPress();
  });
});
