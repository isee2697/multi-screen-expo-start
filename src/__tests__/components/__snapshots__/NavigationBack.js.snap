// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBack /> renders correctly 1`] = `
<TouchableOpacity
  accessibilityComponentType="button"
  accessibilityLabel="go back"
  accessibilityTraits="button"
  accessible={true}
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Object {
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <SvgCircleLeft
    active={true}
    size={24}
  />
</TouchableOpacity>
`;
